client/cli/src/commands/run_cmd.rs

```diff
@@ -250,6 +255,18 @@ pub struct RunCmd {
 		conflicts_with_all = &[ "sentry", "public-addr" ]
 	)]
 	pub sentry_nodes: Vec<MultiaddrWithPeerId>,
+
+	/// Run a temporary node.
+	///
+	/// A random temporary directory will be created to store the configuration and will be deleted
+	/// at the end of the process.
+	///
+	/// Note: the directory is random per process execution.
+	#[structopt(long)]
+	pub tmp: bool,
+
+	#[structopt(skip)]
+	temp_base_path: RefCell<Option<TempDir>>,
```

Comment:
Nope. This is a really bad hack.

Reply:
What part you don't like exactly? (so I can find another way)

Reply:
Not having the temp base path as parameter here.

We could create some custom type that we return in `base_path`

Reply:
Ok done! Now the TempDir is hold in the configuration object instead of the SubstrateCli object, then it's move to the service and dropped with it. Also the feature itself has moved from sc-cli to sc-service.